src/components/node/Node.jsx

```diff
@@ -96,7 +97,22 @@ export default class Node extends React.Component {
                     {this.props.label}
                 </text>
             );
-            node = <image {...nodeProps} href={this.props.svg} width={width} height={height} />;
+
+            // if a view generator is set, it takes precedence over any svg image url
+            if (this.props.viewGenerator) {
+                const nodeInfo = this.props.data;
```

Comment:
I think this is the only thing that I don't like regarding the `viewGenerator` API. Why can't we just pass `this.props` to the custome view generator? That way we would have access to all data (even if it's not used it's ok to have it there). What do you think of removing this more rigid API with the `data` object and simply pass `this.props`?

Reply:
I had my doubts whether to do this or do it like you said at the beginning, but I saw that `this.props` contained a lot of auto-generated data such as the node coordinates or inherited settings from the global config object that were most likely to never be used by the view components. 

What `data` contains is basically the same node object that was passed by the user to the `Graph` component's `data` prop. I thought all of the information the user will be willing to render is the same information they had in their initial data objects, and it's very unlikely they will be using any of the auto-generated props. 

This is the reason why I thought it may be cleaner to do it this way, but if you'd rather pass everything down, I have no problem in changing it. :+1: